components/server/src/server.rs

```diff
@@ -400,6 +423,34 @@ impl<ER: RaftEngine> TiKvServer<ER> {
         ConfigController::new(config)
     }
 
+    /// Run a TiKV server in recovery mode
+    /// recovery mode include:
+    /// 1. no election happen between raft group
+    /// 2. peer valid during a recovery time even without leader in its region
+    /// 3. PD can not put any peer into tombstone
+    /// 4. must ensure all region data with ts less than backup ts (below commit index) are safe
+    fn enter_recovery_mode(
+        config: &mut TiKvConfig,
+    ) {
+        // TOOD: if we do not have to restart TiKV, then, we need exit the recovery mode and config the following parameter back
+        // disable some configuration to ban raft election etc.
+        // For block-level recovery, no raft peers can start a new election.
+        let bt = config.raft_store.raft_base_tick_interval.0;
+        config.raft_store.raft_election_timeout_ticks = MAX_ELECTION_TIME;
+        // time to check if peer alive without the leader, will not check peer during this time interval
+        config.raft_store.peer_stale_state_check_interval = ReadableDuration(bt * 4 * MAX_ELECTION_TIME as _);
+        // duration allow a peer alive without leader in region, otherwise report the metrics and show peer as abnormal
+        config.raft_store.abnormal_leader_missing_duration = ReadableDuration(bt * 4 * MAX_ELECTION_TIME as _);
+        // duration allow a peer alive without leader in region, otherwise report the PD and delete itself(peer)
+        config.raft_store.max_leader_missing_duration = ReadableDuration(bt * 4 * MAX_ELECTION_TIME as _);
+
+        // Disable region split during recovering
+        config.coprocessor.region_max_size = Some(ReadableSize::gb(MAX_REGION_SIZE));
+        config.coprocessor.region_split_size = ReadableSize::gb(MAX_REGION_SIZE);
+        config.coprocessor.region_max_keys = Some(MAX_SPLIT_KEY);
+        config.coprocessor.region_split_keys = Some(MAX_SPLIT_KEY);
```

Comment:
It's better to stop GC, too.

Reply:
And we can also disable rocksdb auto compaction, too.

Reply:
GC already stopped.
BR set recovery mode, start tikv, stop GC and then force leader and delete data.

Reply:
added disable auto compaction